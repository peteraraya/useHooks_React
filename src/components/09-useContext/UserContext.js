import { createContext } from 'react';

export const UserContext = createContext(null);


// Creaci√≥n de este componente 
