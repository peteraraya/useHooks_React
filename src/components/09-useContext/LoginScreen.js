import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    // obtener la referencia al contex
    // extraer el user de ese context 
    const { setUser } = useContext(UserContext);
    /*
        {
            id:123
            name:'Peter'
        }

    */
    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <button
                className="btn btn-primary"
                onClick={ ()=> setUser({
                    id:123,
                    name:'Peter'
                })}
            >
                Login
            </button>
        </div>
    )
}
