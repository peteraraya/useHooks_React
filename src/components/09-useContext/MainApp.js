import React, { useState } from 'react'
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainApp = () => {

    const [user, setUser] = useState({});

    return (           
        <UserContext.Provider value={ {
            // regreso un objeto que tendremos nuestro usery setUser
            user,
            setUser,
        }}>

            <AppRouter />

        </UserContext.Provider>
    )
}
