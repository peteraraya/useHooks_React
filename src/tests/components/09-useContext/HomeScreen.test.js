import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';
import '@testing-library/jest-dom';

describe('Pruebas en <HomeScreen />', () => {

    const user ={
        name:'Pedro',
        email:'piteraraya@gmail.com'
    }
    
    const wrapper = mount(
        <UserContext.Provider  value={{
            user
        }}>
            <HomeScreen /> 
        </UserContext.Provider>
    
    );

    test('debe de mostrarse correctamente', () => {
    
        expect( wrapper ).toMatchSnapshot();
    
    });



    
})

// el shallow hace una renderizaci√≥n solo del componente principal que estoyy probando

// mount nos sirve para renderizar todo en reemplazo de shallow