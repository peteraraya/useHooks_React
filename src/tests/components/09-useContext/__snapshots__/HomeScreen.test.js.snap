// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Pedro",
   "email": "piteraraya@gmail.com"
}
  </pre>
</div>
`;
